rotkehlchen/data_migrations/migrations/migration_20.py

```diff
@@ -145,4 +157,131 @@ def _fix_swap_event_identifiers(rotki: 'Rotkehlchen') -> None:
+            )
+        except sqlcipher.DatabaseError as e:  # pylint: disable=no-member
+            old_db_conn.close()
+            log.error(f'failed to decrypt backup database. wrong password or corrupted file due to {e!s}')  # noqa: E501
+            return
+
+        try:  # create safety backup of current database before making changes
+            current_backup = db.create_db_backup()
+            safety_backup_path = db.user_data_dir / f'{ts_now()}_pre_recovery_v48.backup'
+            shutil.move(src=current_backup, dst=safety_backup_path)
+            log.debug(f'created safety backup at {safety_backup_path}')
+        except OSError as e:
+            log.error(f'failed to create safety backup: {e}. aborting recovery')
+            return
+
+        # recover manual trades (those without a link field) from the backup
+        recovered_events = []
+        with old_db_conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT timestamp, location, base_asset, quote_asset, type, '
+                "amount, rate, fee, fee_currency, link, notes FROM trades WHERE link=''",
+            )
+            # the entirety of this logic is an excerpt from rotkehlchen/db/upgrades/v47_v48.py
+            for row in cursor:
+                spend, receive = get_swap_spend_receive(
+                    is_buy=row[4] in {'A', 'C'},  # A/C = buy/settlement buy; B,D = sell/settlement sell  # noqa: E501
+                    base_asset=Asset(row[2]),
+                    quote_asset=Asset(row[3]),
+                    amount=FVal(row[5]),
+                    rate=Price(FVal(row[6])),
+                )
+                recovered_events.extend(create_swap_events(
+                    timestamp=(timestamp := ts_sec_to_ms(row[0])),
+                    location=(location := Location.deserialize_from_db(row[1])),
+                    spend=spend,
+                    receive=receive,
+                    fee=AssetAmount(
+                        asset=Asset(row[8]),
+                        amount=FVal(row[7]),
+                    ) if row[8] is not None and row[7] is not None else None,
+                    location_label=None,
+                    event_identifier=create_event_identifier_from_swap(
+                        location=location,
+                        timestamp=timestamp,
+                        spend=spend,
+                        receive=receive,
+                        unique_id=row[9],
+                    ),
+                    spend_notes=row[10],
+                ))
+
+        old_db_conn.close()
+        with db.conn.write_ctx() as write_cursor:
+            write_cursor.execute(
+                f'DELETE FROM history_events WHERE event_identifier IN ({placeholders})',
+                location_hashes,
+            )
+            log.debug(f'removed {affected_events_count} incorrectly imported events')
+
+            write_cursor.executemany(
```

Comment:
Would it not be better to do a for loop of insertions and log details of every trade that did not make it before discarding it? This way you just ignore any failures.